src/scripts/main.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
-// write code here
+const fullList = document.querySelectorAll('li');
+
+function sortList(list) {
+  return [...list].sort((a, b) => {
+    const salaryA = +(a.dataset.salary.slice(1).split('')
+      .filter(item => item !== ',').join(''));
+    const salaryB = +(b.dataset.salary.slice(1).split('')
+      .filter(item => item !== ',').join(''));
+
+    return salaryB - salaryA;
+  });
+}
+
+const sortedList = sortList(fullList);
+
+document.querySelector('ul').innerHTML = sortedList.map(item => {
+  return item.outerHTML;
+}).join('');
+
+function getEmployees(list) {
+  const employees = [];
+
+  [...list].map(element => {
+    const employee = {
+      name: element.innerText.trim(),
+      position: element.dataset.position,
+      salary: element.dataset.salary,
+      age: element.dataset.age,
+    };
+
+    employees.push(employee);
+  });
```

Comment:
`map` returns new array, so you don't need to push items in `employees`, just return `employee` in `map` callback and return result of `map` in `getEmployees`